### ISSUE TITLE
sound library?

### ISSUE THREAD

**Issue by bramtayl** [NetHack/NetHack#388]
Maybe the GUI versions could ship with a library of sounds? There's all kinds of sound effects available at freesound.org. In general, sounds seem a bit nicer than messages, in that you don't have to bother reading them.

**Comment by copperwater** [comment]
This would indeed be a windowport thing, since the standard tty interface isn't capable of playing sounds (but it'd be necessary for the core game to provide a new windowport hook to tell it to play a sound file). It could perhaps be done with a system similar to MSGTYPE that would let you associate a sound filename with a message in your config file and play that file whenever a matching message is shown.

Ultimately though I'm not sure that enough players would bother to use such a feature, and I think it's unlikely nethack would ship with the sound files themselves - at most, there'd be a make target to download a number of standard songs from freesound or wherever.

**Comment by ostrosablin** [comment]
Actually, Vulture for NetHack has attempted to bring sounds into the game (probably by hardcoding sounds to specific messages). It may just be me, but sounds seem good for several first games, and after that they become an annoyance, because they're repetitive, and don't really add anything to game. It's not that experience becomes more "real" or that it makes game world more immersive.

Yeah, of course, this system could be improved, several variations of sound could be added and sounds can even be slightly altered programmatically (e.g. tone and speed) to make it feel like each sound is unique, but for me, "you hear" messages feel much more natural in NetHack, than attempts to replace them with sounds.

**Comment by paxed** [comment]
There's the SOUNDS configuration option: https://nethack.org/v366/Guidebook.html#toc_9.12

**Comment by bramtayl** [comment]
Yes, I had figured out how to mess around with the nethackrc file to do this myself, but I figured it might be nice to share

**Comment by bramtayl** [comment]
Someone has already done a lot of work here: http://xoomer.virgilio.it/pid. I had to mess around with the regular expressions but it seems to mostly work.

**Comment by bramtayl** [comment]
This is the julia code I used to update their options file:

```julia
open("/home/brandon/Downloads/pattern.txt") do input
    open("/home/brandon/.nethackrc", append = true) do output
        println(output)
        for line in eachline(input)
            # ignore the first line; make sure to specify the sound dir later
            if !isempty(line) && line != "SOUNDDIR=sounds"
                # escape periods, but only if they are not the period in .wav
                # replace * with .*
                edited = replace(replace(line, r"\.(?!wav)" => raw"\."), "*" => ".*")
                println(output, edited)
                # suppress the message (because users hear the sound already)
                print(output, "MSGTYPE=hide \"")
                # the message is the first quote in the line
                print(output, match(r"\"([^\"]*)\"", edited).captures[1])
                println(output, '\"')
            end
        end
    end
end
```

**Comment by nhmall** [comment]
This issue was opened for a 3.6 version of NetHack that is no longer being actively maintained. 

Please try the NetHack-5.0.0 release, which has had some sound support added for some platforms.
If you are still dissatisfied, please open and describe the issue and specify the NetHack version in the opening comment.
Thank you.
